### PR TITLE
Server landingpage custom prefix

### PR DESCRIPTION
**Pull request by elpaso**
Add a `QGIS_SERVER_LANDING_PAGE_PREFIX` server setting:

Prefix of the path component of the landing page base URL, default is empty (since QGIS 3.20).

### CODE REVIEW

**Comment by elpaso** [comment]
@rduivenvoorde can you test this?

**Comment by rduivenvoorde** [comment]
> @rduivenvoorde can you test this?

Yep, but not on short notice...
Maybe @rosaguilar can sooner/earlier?

**Comment by rduivenvoorde** [comment]
Ok, testing some configurations here:

using 
```export QGIS_SERVER_LANDING_PAGE_PROJECTS_DIRECTORIES="/home/richard/z/21/qgis/qgisserver/"```
OR setting it in apache conf

1) running qgis_mapserver as standalone on port 8000: if I set '/qgiz' as landing prefix using

```export QGIS_SERVER_LANDING_PAGE_PREFIX="/qgiz"```

I do get the landing page (html and osm background maps), but the /qgiz seam not to be picked in the javascript? I do not get my local data layers.
In debugging window of browser I see it is trying to reach:

http://localhost:8000/project/5186a2e5f67c54c2af68f9667039c33d/?&service=WMS&request=GetMap&version=1.1.1&layers=postgis&styles=&format=image%2Fpng&transparent=true&width=400&height=320&srs=EPSG%3A3857&bbox=-371789.7055790973%2C6051366.655280832%2C1584998.2185214148%2C7616796.994561241

at that moment, but it should put /qgiz between http://localhost:8000/ and project/ ?

2) running the same configuration, BUT behind an apache reverse proxy:

```
  <Location "/qgiz/">
        ProxyPass "http://localhost:8000/qgiz/"
        ProxyPassReverse "http://localhost:8000/qgiz/"
  </Location>
```

Same result: http://localhost/qgiz/ is showing landingpage and osm, but not my local layers because it is not using the landing page prefix in the (apparently) non relative url's?

BUT!! If i click on one of the project tiles on the Browse link: THEN I get my local files (but not the osm anymore, which is fine I think)

3) only using QGIS_SERVER_LANDING_PAGE_PREFIX in apache conf:

```
FcgidInitialEnv QGIS_SERVER_LANDING_PAGE_PREFIX "/qgiz"

```
Not sure what the landingpage then is supposed to be:
http://localhost/qgiz/ returns a not found
http://localhost/cgi-bin/qgis_mapserv.fcgi/qgiz returns a "http://localhost/cgi-bin/qgis_mapserv.fcgi/qgiz"

4) NOT setting the landing page: and running qgis_mapserver on port 8000 all is fine: on the landing page both osm and local layers, and after clicking browse on a project-tile I see the local layers.

Not 100% if these are code issues or more configuration/knowledge issues though....

**Comment by rosaguilar** [comment]
> > @rduivenvoorde can you test this?
> 
> Yep, but not on short notice...
> Maybe @rosaguilar can sooner/earlier?

I am happy to test but not sure how?

**Comment by elpaso** [comment]
@rduivenvoorde thank you for testing, I fixed the relative WMS urls in catalog view, it should be ok now.

File: src/server/services/landingpage/qgslandingpagehandlers.h [hunk 1]
```diff
@@ -54,6 +54,13 @@ class QgsLandingPageHandler: public QgsServerOgcApiHandler
     QgsServerOgcApi::Rel linkType() const override { return QgsServerOgcApi::Rel::self; }
     const QString templatePath( const QgsServerApiContext &context ) const override;
 
+    /**
+     *  Returns the path prefix, default is empty. Also makes sure that not-empty
+     *  prefix starts with "/" (ex: "/mylandingprefix"
```
Comment (pblottiere) [inline src/server/services/landingpage/qgslandingpagehandlers.h:59]:
```suggestion
     *  prefix starts with "/" (ex: "/mylandingprefix")
```

File: src/server/services/landingpage/qgslandingpagehandlers.h [hunk 2]
```diff
@@ -54,6 +54,13 @@ class QgsLandingPageHandler: public QgsServerOgcApiHandler
...
+     */
```
Comment (pblottiere) [inline src/server/services/landingpage/qgslandingpagehandlers.h:60]: I know we don't have a lot of `\since` in server code but we probably should :).

Comment (elpaso) [reply src/server/services/landingpage/qgslandingpagehandlers.h:60]: It's not part of the API.

Comment (elpaso) [reply src/server/services/landingpage/qgslandingpagehandlers.h:60]: Maybe add it to the settings description if I didn't already.

**Pull request merged**